### PR TITLE
Add COESSING hub

### LINKED ISSUE #1516: [New Hub] COESSING Pangeo-Style Hub
### Hub Description

This is a request for a new hub to support the  upcoming [Coastal Environment Summer School in Ghana](https://coessing.org/). The school runs August 1-5, 2022 so this is a hot request. 2i2c supported last year's version of this event. The 2022 edition will take place in Nigeria. This summer school is scoped under the Columbia Moore grant led by @rabernat.

### Community Representative(s)

- @paigem

### Important dates

- **Target start date**: July 25, 2022
- **Required start date**:  July 30, 2022
- **Any important dates for usage**: August 1-5, 2022 is when the school takes place.

### Hub Authentication Type

GitHub Authentication (e.g., @mygithubhandle)

### Hub logo information

- **URL to Hub Image**: {{ URL HERE }}
- **URL for Image Link**: {{ URL HERE }}

### Hub user image

- Repository for user image: { REPO LINK IF IT EXISTS }
- User image registry: { REGISTRY IF ONE ALREADY EXISTS }
- User image tag and name: { NAME AND TAG IF IT EXISTS }

### Extra features you'd like to enable

- [ ] Specific cloud provider or datacenter: 
- [ ] Dedicated Kubernetes cluster
- [ ] Scalable Dask Cluster

### Other relevant information

Email with information shared by Paige: 

Hi 2i2c!

I am hoping that we can set up another Hub for an oceanography summer school in West Africa ([COESSING](https://coessing.org/)) in a couple weeks. The school will be held in hybrid format with an in-person portion in Lagos, Nigeria, and a virtual component with primarily West African participants. We used a 2i2c Hub last year for our entirely virtual school and it worked great!

Logistics:

    Workshop dates: Monday, August 1 - Friday, August 5
    Desired start date of Hub: Monday, July 25 (I realize this is very soon - please let me know if this is not possible)
    Desired end date of Hub: August 31
    Estimated maximum usage at any given point in time (likely only for ~1 hour each day): 120 people
    Desired authentication: Google accounts that are added manually (i.e. I will add them manually as the admin)

Other details:
A few desired functions, partly based on a recent discussion with Chris Holdgraf:

    Ideally this Hub is a “sub” version of the Pangeo Hub. The idea is that we will invite any participants who are computationally inclined to join the Pangeo Hub after the workshop. However, because we may have up to 200 participants engage with content on the JupyterHub, we would like to have our own dedicated Hub for the workshop.
    We would like the Pangeo Cloud computing environment, with the option to add more packages on top.
    Last year we used the Pangeo Google Cloud distribution on us-central1. This year we have one instructor who wishes to use aws-us-west-2 to access NASA’s PODAAC, but we would still like access to datasets that (I believe) are stored on Google Cloud (eg ECCOv4r3). Is there a way to have both Google Cloud and AWS as an option? If that is too complicated/expensive, then let’s stick with Google Cloud.
    This Hub will be Python-based. However, is it possible to include a Julia kernel as an option on this Hub?
    We will use very little (if any) Dask capability. Is this something we can add closer to the time? I am still communicating with instructors to determine if Dask will be needed.
    I have discussed possible funding options with Chris Holdgraf - please let me know if 2i2c does not have funding for this Hub, and Ryan Abernathey has offered some funding to cover this.

Please let me know if you have any more questions! I look forward to working with 2i2c again!
Paige

### Hub URL

<hub ID>.<hub cluster>.2i2c.cloud

### Hub Type

daskhub

### Tasks to deploy the hub

- [x] Engineer who will deploy the hub is assigned
- [x] Deploy information filled in above
- [x] Initial Hub deployment PR: <link to PR>
- [x] Administrators able to log on
- [ ] Community Representative satisfied with hub environment
- [ ] Hub now in steady-state

### PR DESCRIPTION
**Pull request by yuvipanda**
Fixes #1516

### CODE REVIEW

File: helm-charts/images/hub/Dockerfile [hunk 1]
```diff
@@ -12,6 +12,13 @@ ENV CONFIGURATOR_VERSION ed7e3a0df1e3d625d10903ef7d7fd9c2fbb548db
 
 RUN pip install --no-cache git+https://github.com/yuvipanda/jupyterhub-configurator@${CONFIGURATOR_VERSION}
 
+# Needed to install JupyterHub from source
+USER root
+RUN apt update --yes > /dev/null && apt install --yes --no-install-recommends nodejs npm > /dev/null
+USER $NB_USER
+# Until https://github.com/jupyterhub/jupyterhub/pull/3984 is merged
+RUN pip install --upgrade git+https://github.com/meeseeksmachine/jupyterhub@72e4119e1a9f5fead690cb399dba36143d6d2ecc
+
```
Comment (damianavila) [inline helm-charts/images/hub/Dockerfile:21]: Installing from source is always an additional risk.
When this is merged, let's create an issue to go back to a released version ASAP.

File: config/clusters/pangeo-hubs/coessing.values.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+basehub:
+  userServiceAccount:
+    annotations:
+      # iam.gke.io/gcp-service-account: pangeo-hubs-prod@pangeo-integration-te-3eea.iam.gserviceaccount.com
```
Comment (sgibson91) [inline config/clusters/pangeo-hubs/coessing.values.yaml:4]: Do you need to replace this annotation?

**Review by sgibson91 — APPROVED** [review]
One comment but LGTM

File: config/clusters/pangeo-hubs/coessing.values.yaml (hunk 2 above)
Comment (yuvipanda) [reply config/clusters/pangeo-hubs/coessing.values.yaml]: Yep! Just done, and I had to run terraform too!

**Review by GeorgianaElena — APPROVED** [review]
🚀

**Comment by sgibson91** [comment]
I [`599ee61` (#1537)](https://github.com/2i2c-org/infrastructure/pull/1537/commits/599ee61e8291228ccb7fe3626d3e543c30fb5a97) which addresses the change of logo request made here: https://github.com/2i2c-org/infrastructure/issues/1516#issuecomment-1194791536 I haven't deployed it yet because my laptop is acting up and as soon as the helm upgrade starts, everything slows right down and processes just start hanging.

We should try to get this into a mergeable state though. @yuvipanda what action points remain that would make this PR no longer a draft?

**Comment by yuvipanda** [comment]
Thanks a lot for taking care of this, @sgibson91! This mostly needs gh-scoped-creds to be setup, but I'll get this merged for now!

**Pull request merged**